knotx-core/knotx-template-engine/src/main/java/com/cognifide/knotx/engine/impl/TemplateSnippetProcessor.java

```diff
@@ -51,10 +53,10 @@ public TemplateSnippetProcessor(Vertx vertx, TemplateEngineConfiguration configu
 
     public Observable<String> processSnippet(final HtmlFragment fragment, TemplateEngineRequest request) {
         return fragment.getServices()
-                .filter(serviceEntry -> serviceEntry.canServeMethodType(request.getServerRequestMethod()))
+                .filter(serviceEntry -> serviceEntry.canServeRequest(fragment, request))
                 .doOnNext(this::traceService)
                 .flatMap(serviceEngine::findServiceLocation)
-                .flatMap(serviceItem -> serviceEngine.doServiceCall(serviceItem, request.getHeaders(), request.getServerRequestMethod()),
+                .flatMap(serviceItem -> serviceEngine.doServiceCall(serviceItem, request.getHeaders(), computeServiceMethodType(request, serviceItem.getMethodType()), request.getFormAttributes()),
```

Comment:
Can we move functionality of computeServiceMethodType() method to the ServiceEngine level, to have achieve such invocation:
```
flatMap(serviceItem -> serviceEngine.doServiceCall(serviceItem, request))
```
inside method you can get:
- headers from request
- formAttributes from request
- methodType from serviceItem

This way it's going to be more readable and fits better to serviceEngine as he suppose to know how to make a call given some data.